src/flick/friend/views.py

```diff
@@ -105,6 +105,9 @@ def _update_notification(self, from_user, to_user):
         if not notif_exists:
             return
         notif = Notification.objects.get(notif_type="friend_request", from_user=from_user_old, to_user=to_user_old)
+        notif.from_user = to_user_old
```

Comment:
this is super clever!!!! however, one thing is that lucy and i recently added this feature #216 where we want to mark if notifications have been viewed, and the way we do this is lucy sends me a timestamp of when the user viewed their notifications screen, and whenever i need to calculate `num_notifs` in `/me`, i only count notifications that have been created after the timestamp. so, you will need to change the `notification/models.py` to introduce an `updated_at` field to notification models.py, and here's an example
```
created_at = models.DateTimeField(auto_now_add=True)
updated_at = models.DateTimeField(auto_now=True)
```
there's a difference if you notice hehe :) so then you can change in `user/models.py`
```
unviewed_notifs = notifs.filter(Q(created_at__gt=self.notif_time_viewed)) if self.notif_time_viewed else notifs
```
to
```
unviewed_notifs = notifs.filter(Q(updated_at__gt=self.notif_time_viewed)) if self.notif_time_viewed else notifs
```